sha-crypt/src/b64.rs

```diff
@@ -1,64 +1,64 @@
 //! Base64 encoding support
 
-use crate::defs::{MAP, TAB};
+use crate::defs::{
+    BLOCK_SIZE_SHA256, BLOCK_SIZE_SHA512, MAP_SHA256, MAP_SHA512, PW_SIZE_SHA256, PW_SIZE_SHA512,
+};
 use alloc::vec::Vec;
+use base64ct::{Base64ShaCrypt, Encoding};
 
 #[cfg(feature = "simple")]
 use crate::errors::DecodeError;
 
-pub fn encode(source: &[u8]) -> Vec<u8> {
-    let mut out: Vec<u8> = vec![];
-    for entry in &MAP {
-        let mut w: usize = 0;
-        if entry.3 > 2 {
-            w |= source[entry.2 as usize] as usize;
-            w <<= 8;
-            w |= source[entry.1 as usize] as usize;
-            w <<= 8;
-        }
-        w |= source[entry.0 as usize] as usize;
+pub fn encode_sha512(source: &[u8]) -> Vec<u8> {
```

Comment:
FWIW it'd be really great to get rid of `Vec` here if possible. This is one of the few crates we have which has a hard dependency on `alloc`.

Reply:
I don't have experience in no-std libraries.  I'll see what I can do, but I won't spend too much effort on this.

Reply:
Here it should be fairly easy to get rid of by just returning the array, rather than converting it into a `Vec`

Reply:
The other vectors look harder to eliminate though.

Reply:
No worries then!